library/data/component/src/main/java/org/quiltmc/qsl/component/impl/predicates/FilteredInheritedInjectionPredicate.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.component.impl.predicates;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class FilteredInheritedInjectionPredicate extends InheritedInjectionPredicate {
+	private final List<Class<?>> exceptions;
+
+	public FilteredInheritedInjectionPredicate(Class<?> clazz, Class<?>[] exceptions) {
+		super(clazz);
+		this.exceptions = new ArrayList<>(Arrays.asList(exceptions));
```

Comment:
```suggestion
		this.exceptions = Arrays.asList(exceptions);
```
`Arrays.asList` returns an immutable list. I'm not sure why you'd want this predicate to be mutable, especially since all the other predicates seem to be immutable.